span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanPreProcessor.java

```diff
@@ -40,13 +47,25 @@ public JaegerSpanPreProcessor() {
   JaegerSpanPreProcessor(Config jobConfig) {
     tenantIdHandler = new TenantIdHandler(jobConfig);
     spanFilter = new SpanFilter(jobConfig);
+    lateArrivalThresholdDuration = configureLateArrivalThreshold(jobConfig);
   }
 
   @Override
   public void init(ProcessorContext context) {
     Config jobConfig = (Config) context.appConfigs().get(SPAN_NORMALIZER_JOB_CONFIG);
     tenantIdHandler = new TenantIdHandler(jobConfig);
     spanFilter = new SpanFilter(jobConfig);
+    lateArrivalThresholdDuration = configureLateArrivalThreshold(jobConfig);
+  }
+
+  private Duration configureLateArrivalThreshold(Config jobConfig) {
+    Duration configuredThreshold = jobConfig.getDuration(LATE_ARRIVAL_THRESHOLD_CONFIG_KEY);
+    Duration minThreshold = Duration.of(30, ChronoUnit.SECONDS);
```

Comment:
This was suggested by @laxman-traceable . I think, he has suggestions for max value too. But, I have kept it open.